src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1293,11 +1294,9 @@ private static BoundNode Rewrite(ImmutableDictionary<BoundExpression, (Nullabili
         internal static bool NeedsAnalysis(CSharpCompilation compilation)
         {
 #if DEBUG
-            // Always run analysis in debug builds so that we can more reliably catch
-            // nullable regressions e.g. https://github.com/dotnet/roslyn/issues/40136
-            return true;
+            return compilation.ShouldRunNullableAnalysisAndIgnoreResults;
```

Comment:
>return compilation.ShouldRunNullableAnalysisAndIgnoreResults; [](start = 12, length = 61)

For clarity and consistency with other use-sites, I think this condition should be combined with ```!canSkipAnalysis```